drivers/bluetooth/hci/rpmsg.c

```diff
@@ -317,9 +325,42 @@ static int bt_rpmsg_open(void)
 	return 0;
 }
 
+static int bt_rpmsg_close(void)
+{
+	int err;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_RESET, NULL, NULL);
+	if (err) {
+		LOG_ERR("Sending reset command failed with: %d", err);
+		return err;
+	}
+
+	err = ipc_service_deregister_endpoint(&hci_ept);
+	if (err) {
+		LOG_ERR("Deregistering HCI endpoint failed with: %d", err);
+		return err;
+	}
+
+	const struct device *hci_ipc_instance =
+		DEVICE_DT_GET(DT_CHOSEN(zephyr_bt_hci_rpmsg_ipc));
+
+	err = ipc_service_close_instance(hci_ipc_instance);
+	if (err) {
+		LOG_ERR("Closing IPC service failed with: %d", err);
+		return err;
+	}
+
+#ifdef CONFIG_SOC_NRF5340_CPUAPP
+	cpunet_forceoff();
```

Comment:
Thinking about it, this might be a problem if other stuff's running on the network core, like say zigbee or a custom protocol. They'll just get an abrupt communication link break.

What I propose is that this gets merged, and then I'll make another PR, specific to 53, that enables/disables the forcing on and off of the netcore.

That way advanced users that have something else on the netcore can just disable that.

Reply:
I think that we can go that way.

In the situation you are describing I see another problem - the `bt_rpmsg_close(void)` do not know about another IPC endpoints so if there is any opened the `ipc_service_close_instance(hci_ipc_instance)` will fail.